PYME/Acquire/Hardware/pco/pco_cam.py

```diff
@@ -68,25 +74,21 @@ def CamReady(self):
         return self.initalized
 
     def ExtractColor(self, chSlice, mode):
-
         # Somehow this check matters... shouldn't this be taken care of by ExpReady???
-        if self.GetNumImsBuffered() < 1:
+        if (not self.recording) or (self.GetNumImsBuffered() < 1):
             return True
```

Comment:
The `return True` here is super unsafe (or will at the very least result in data corruption) - because we return our result by writing into the `chslice` input argument, returning from this method with no exception set is implicitly stating that you have put a new frame into `chslice`. If you are in this method and there is no data available, the only safe thing to do is to raise an exception.

Reply:
I know this is not new in this PR - but thought it worth flagging for debug efforts in general.

Reply:
Makes sense. I'll push a commit fixing this in a second.